### PR TITLE
Additional test cases for PostHog implementation - Event properties

### PR DESCRIPTION
**Pull request by milagi**
# Checklist

- [ ] [JIRA](https://input-output.atlassian.net/browse/LW-8324)
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

Explain how does this PR solves the problem stated in JIRA ticket.
You can also enumerate different alternatives considered while approaching this task.

## Testing

Describe here, how the new implementation can be tested.
Provide link or briefly describe User Acceptance Criteria/Tests that need to be met

## Screenshots

Attach screenshots here if implementation involves some UI changes

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/1939/6097818966/index.html) for [49b342dd](https://github.com/input-output-hk/lace/pull/494/commits/49b342ddfd74a1cc3231157a3d667e6b6e1bb2c5)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 28     | 4      | 0       | 0     | 32    | ❌     |

### CODE REVIEW

File: packages/e2e-tests/src/features/analytics/AnalyticsEventProperitesExtended.feature [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  Given Wallet is synced
+
+  @LW-7703
+  Scenario: Analytics - Verify event properties
+    Given I set up request interception for posthog analytics request(s)
+    And I am on NFTs extended page
+    And I validate that event has correct properties
+
+  @LW-8349
+  Scenario: Extended View - Analytics - Verify event properties - Send - Send button
```
Comment (bslabiak) [inline packages/e2e-tests/src/features/analytics/AnalyticsEventProperitesExtended.feature:13]: Please use Analytics as first in Scenario name.

File: packages/e2e-tests/src/features/analytics/AnalyticsEventProperitesExtended.feature [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    Given I set up request interception for posthog analytics request(s)
+    When I click "Send" button on page header
+    Then I validate that the event includes "trigger_point" property
+    And I validate that the "send | send | click" event includes property "trigger_point" with value "send button" in posthog
+
+  @LW-8350
+  Scenario: Extended View - Analytics - Verify event properties - Send - Send NFT button
```
Comment (bslabiak) [inline packages/e2e-tests/src/features/analytics/AnalyticsEventProperitesExtended.feature:20]: Please use Analytics as first in Scenario name.

File: packages/e2e-tests/src/features/analytics/AnalyticsEventProperitesExtended.feature [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    Given I set up request interception for posthog analytics request(s)
+    And I am on NFTs extended page
+    And I left click on the NFT with name "Ibilecoin_1" on NFTs page
+    When I click "Send NFT" button on NFT details drawer
+    Then I validate that the event includes "trigger_point" property
+    And I validate that the "send | send | click" event includes property "trigger_point" with value "nfts page" in posthog
+
+  @LW-8351
+  Scenario: Extended View - Analytics - Verify event properties - Send - Send token
```
Comment (bslabiak) [inline packages/e2e-tests/src/features/analytics/AnalyticsEventProperitesExtended.feature:29]: Please use Analytics as first in Scenario name.

File: packages/e2e-tests/src/features/analytics/AnalyticsEventProperitesPopup.feature [hunk 4]
```diff
@@ -0,0 +1,28 @@
+@NFTs-Popup @Testnet @Mainnet
+Feature: Analytics - Posthog - Event properties
+
+  Given Wallet is synced
+
+  @LW-8352
+  Scenario: Popup View - Analytics - Verify event properties - Send - Send button
```
Comment (bslabiak) [inline packages/e2e-tests/src/features/analytics/AnalyticsEventProperitesPopup.feature:7]: Please use Analytics as first in Scenario name.

File: packages/e2e-tests/src/features/analytics/AnalyticsEventProperitesPopup.feature [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    Given I set up request interception for posthog analytics request(s)
+    When I click "Send" button on Tokens page in popup mode
+    Then I validate latest analytics single event "send | send | click"
+    And I validate that the "send | send | click" event includes property "trigger_point" with value "send button" in posthog
+
+  @LW-8353
+  Scenario: Popup View - Analytics - Verify event properties - Send - Send NFT button
```
Comment (bslabiak) [inline packages/e2e-tests/src/features/analytics/AnalyticsEventProperitesPopup.feature:14]: Please use Analytics as first in Scenario name.

File: packages/e2e-tests/src/features/analytics/AnalyticsEventProperitesPopup.feature [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    Given I set up request interception for posthog analytics request(s)
+    And I am on NFTs popup page
+    And I left click on the NFT with name "Ibilecoin_1" on NFTs page
+    When I click "Send NFT" button on NFT details drawer
+    Then I validate latest analytics single event "send | send | click"
+    And I validate that the "send | send | click" event includes property "trigger_point" with value "nfts page" in posthog
+
+  @LW-8354
+  Scenario: Popup View - Analytics - Verify event properties - Send - Send token
```
Comment (bslabiak) [inline packages/e2e-tests/src/features/analytics/AnalyticsEventProperitesPopup.feature:23]: Please use Analytics as first in Scenario name.

File: packages/e2e-tests/src/steps/analyticsSteps.ts [hunk 7]
```diff
@@ -77,3 +77,23 @@ Then(/^I validate that event has correct properties$/, async () => {
...
 });
+
+Then(/^I validate that the event includes "([^"]*)" property$/, async (property: string) => {
+  await browser.pause(1000);
+  const actualEventPayload = await getLatestEventPayload();
+  const expectedProperties = [property];
+  for (const expectedProperty of expectedProperties) {
+    expect(Object.prototype.hasOwnProperty.call(actualEventPayload.properties, expectedProperty)).to.be.true;
+  }
+  expect(Object.prototype.hasOwnProperty.call(actualEventPayload, 'timestamp')).to.be.true;
```
Comment (bslabiak) [inline packages/e2e-tests/src/steps/analyticsSteps.ts:88]: If one property is validated there is no need to run assertions in loop.

File: packages/e2e-tests/src/features/analytics/AnalyticsEventProperitesExtended.feature [hunk 8]
```diff
@@ -0,0 +1,35 @@
+@NFTs-Extended @Testnet @Mainnet
+Feature: Analytics - Posthog - Event properties
+
+  Given Wallet is synced
+
+  @LW-7703
+  Scenario: Analytics - Verify event properties
+    Given I set up request interception for posthog analytics request(s)
+    And I am on NFTs extended page
+    And I validate that event has correct properties
```
Comment (DominikGuzei) [inline packages/e2e-tests/src/features/analytics/AnalyticsEventProperitesExtended.feature:10]: out of curiosity: does this validate the general base properties of our events?

**Review by DominikGuzei — APPROVED** [review]
LGTM 👍

File: packages/e2e-tests/src/features/analytics/AnalyticsEventProperitesExtended.feature (hunk 8 above)
Comment (bslabiak) [reply packages/e2e-tests/src/features/analytics/AnalyticsEventProperitesExtended.feature:10]: it validates existence of following properties, not checking all the values. Usually we are validating specific properties values if there are custom ones or connected with specific features e.g aliases.
timestamp + 
const expectedProperties = [
    '$current_url',
    '$window_id',
    '$browser',
    '$browser_language',
    '$browser_version',
    '$device_type',
    '$host',
    '$insert_id',
    '$lib',
    '$lib_version',
    '$lib_version',
    '$os',
    '$os_version',
    '$pageview_id',
    'posthog_project_id',
    '$pathname',
    '$referrer',
    '$referring_domain',
    '$screen_height',
    '$screen_width',
    '$session_id',
    '$viewport_height',
    '$viewport_width',
    'sent_at_local',
    'view'
  ];

File: packages/e2e-tests/src/steps/analyticsSteps.ts [hunk 9]
```diff
@@ -77,3 +77,20 @@ Then(/^I validate that event has correct properties$/, async () => {
   }
   expect(Object.prototype.hasOwnProperty.call(actualEventPayload, 'timestamp')).to.be.true;
 });
+
+Then(/^I validate that the event includes "([^"]*)" property$/, async (property: string) => {
+  await browser.pause(1000);
+  const actualEventPayload = await getLatestEventPayload();
+  expect(Object.prototype.hasOwnProperty.call(actualEventPayload.properties, property)).to.be.true;
+  expect(Object.prototype.hasOwnProperty.call(actualEventPayload, 'timestamp')).to.be.true;
```
Comment (bslabiak) [inline packages/e2e-tests/src/steps/analyticsSteps.ts:85]: This line can be removed

**Review by bslabiak — APPROVED** [review]

**Pull request merged**